src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs

```diff
@@ -93,5 +97,22 @@ internal static int[] GetCurrencyNegativePatterns(CultureInfo cultureInfo)
 
             throw DateTimeFormatInfoData.GetCultureNotSupportedException(cultureInfo);
         }
+        
+        private static Version GetOSXKernelVersion()
+        {
+            if (s_isOSX)
+            {
+                ulong bytesLength = 256;
+                byte[] bytes = new byte[bytesLength];
+                sysctlbyname("kern.osrelease", bytes, ref bytesLength, null, 0);
+                string versionString = Encoding.UTF8.GetString(bytes);
+                return Version.Parse(versionString);
+            }
+            
+            return new Version(0, 0, 0);
+        }
+        
+        [DllImport("libc")]
+        private static extern int sysctlbyname(string ctlName, byte[] oldp, ref ulong oldpLen, byte[] newp, ulong newpLen);
```

Comment:
I know we've beaten this issue over and over on this thread, but I just wanted to respond to a few points.

> "just asking the OS for the data"

It isn't that simple. On Windows it might be that simple to just call a native method to get the culture data, but on Unix we are using ICU, and then doing our own logic on top of it where necessary.  One such example is https://github.com/dotnet/coreclr/blob/master/src/corefx/System.Globalization.Native/localeNumberData.cpp#L258 where we turn an ICU string number pattern into one of our own enum values. This parsing logic isn't simple, and to @stephentoub's point, if the test had to do the same thing, we wouldn't really be testing anything.  Note there are other cases like this, just poke around under coreclr/src/corefx/System.Globalization.Native to see them.

Building on this point, @eatdrinksleepcode, you bring up a really good idea and is the real way to solve this issue IMO. However, it would be a lot of work to get some sort of stub in place. A good chunk of the code we are testing here is native code built around ICU. We would need to inject the stub lower than that native code in order to keep it under test. Some day we might be able to build something like this, but I just don't think we have the time/resources right now.